addons/mrp/tests/test_manual_consumption.py

```diff
@@ -230,3 +230,36 @@ def get_moves(mo):
         move_auto, move_manual = get_moves(backorder)
         self.assertEqual(move_auto.manual_consumption, False)
         self.assertEqual(move_manual.manual_consumption, True)
+
+    def test_update_manual_consumption_00(self):
+        """
+        Check that the manual consumption is set to true when the quantity is manualy set.
+        """
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.bom_id = self.bom_1
+        mo_form.product_qty = 1
+        mo = mo_form.save()
+        mo.action_confirm()
+        with Form(mo) as fmo:
+            with fmo.move_raw_ids.edit(0) as line_0:
+                line_0.quantity = 3.0
+                line_0.picked = True
+        mo.button_mark_done()
+        self.assertRecordValues(mo.move_raw_ids, [{'quantity': 3.0, 'picked': True}, {'quantity': 1.0, 'picked': True}])
+
+    def test_update_manual_consumption_01(self):
+        """
+        Check that the quantity of a raw line that is manually consumed is not updated
+        when the qty producing is changed and that others are.
+        """
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.bom_id = self.bom_1
+        mo_form.product_qty = 1
+        mo = mo_form.save()
+        mo.action_confirm()
+        with Form(mo) as fmo:
+            with fmo.move_raw_ids.edit(0) as line_0:
+                line_0.quantity = 3.0
+                line_0.picked = True
+            fmo.qty_producing = 2.0
+        self.assertRecordValues(mo.move_raw_ids, [{'quantity': 3.0, 'picked': True}, {'quantity': 2.0, 'picked': True}])
```

Comment:
I would make these tests a little more robust so that they're easier for people to understand in the future by doing the following:

- ensure the components are in stock (with `self.env['stock.quant']._update_available_quantity()`)
- add an assert that the BoM product quantities are reserved after the action_confirm (so others can see what the expected "To Consume" quantities are if it wasn't manually set)
-  add asserts that check that the `manual_consumption=True/False` for the moves are set as expected (in case someone changes the quantities for self.bom_1 which can make it so the test always passes in the end)